src/pages/leaderboard/api/get-leaderboard.ts

```diff
@@ -0,0 +1,73 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { bech32mPositionId } from '@penumbra-zone/bech32m/plpid';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { AssetId } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { PositionId } from '@penumbra-zone/protobuf/penumbra/core/component/dex/v1/dex_pb';
+import { serialize, Serialized } from '@/shared/utils/serializer';
+import { toValueView } from '@/shared/utils/value-view';
+import { getURLParams, LeaderboardData, LeaderboardPageInfo, intervalFilterToSQL } from './utils';
+import { pindexer } from '@/shared/database';
+
+export const GET = async (
+  req: NextRequest,
+): Promise<NextResponse<Serialized<LeaderboardPageInfo | { error: string }>>> => {
+  const chainId = process.env['PENUMBRA_CHAIN_ID'];
+  if (!chainId) {
+    return NextResponse.json({ error: 'Error: PENUMBRA_CHAIN_ID is not set' }, { status: 500 });
+  }
+
+  try {
+    const registryClient = new ChainRegistryClient();
+    const registry = await registryClient.remote.get(chainId);
+
+    const filters = getURLParams(new URLSearchParams(req.nextUrl.search));
+    const result = await pindexer.queryLeaderboard(
+      filters.limit,
+      intervalFilterToSQL[filters.interval],
+      filters.base,
+      filters.quote,
+    );
+
+    const mapped = await Promise.all(
+      result.map(position => {
+        const asset1 = new AssetId({ inner: position.context_asset_start });
+        const asset2 = new AssetId({ inner: position.context_asset_end });
+        const metadata1 = registry.tryGetMetadata(asset1);
+        const metadata2 = registry.tryGetMetadata(asset2);
+
+        if (!metadata1 || !metadata2) {
+          return undefined;
```

Comment:
super minor nitpick, but this can be shortened to just return; - same behaviour.